pjmedia/include/pjmedia/av_sync.h

```diff
@@ -0,0 +1,206 @@
+ */
+typedef struct pjmedia_av_sync pjmedia_av_sync;
+
+
+/**
+ * Media synchronization handle, opaque.
+ */
+typedef struct pjmedia_av_sync_media pjmedia_av_sync_media;
+
+
+/**
+ * Media settings.
+ */
+typedef struct {
+    /**
+     * Name of the media
+     */
+    char                       *name;
+
+    /**
+     * Media clock rate or sampling rate.
+     */
+    unsigned                    clock_rate;
+} pjmedia_av_sync_media_setting;
+
+
+/**
+ * Get default settings for media.
+ *
+ * @param setting           The media setting.
+ */
+PJ_DECL(void) pjmedia_av_sync_media_setting_default(
+                                pjmedia_av_sync_media_setting *setting);
+
+/**
+ * Create media synchronizer.
+ *
+ * @param endpt             The media endpoint.
+ * @param setting           Synchronization setting, must be NULL for now.
+ * @param av_sync           The pointer to receive the media synchronizer.
+ *
+ * @return                  PJ_SUCCESS on success.
+ */
+PJ_DECL(pj_status_t) pjmedia_av_sync_create(
+                                pjmedia_endpt *endpt,
+                                const void *setting,
+                                pjmedia_av_sync **av_sync);
+
+
+/**
+ * Destroy media synchronizer.
+ *
+ * @param av_sync           The media synchronizer.
+ */
+PJ_DECL(void) pjmedia_av_sync_destroy(pjmedia_av_sync* av_sync);
+
+
+/**
+ * Add a media to synchronizer.
+ *
+ * @param av_sync           The media synchronizer.
+ * @param setting           The media setting.
+ * @param av_sync_media     The pointer to receive the media synchronization
+ *                          handle.
+ *
+ * @return                  PJ_SUCCESS on success.
+ */
+PJ_DECL(pj_status_t) pjmedia_av_sync_add_media(
+                                pjmedia_av_sync* av_sync,
+                                const pjmedia_av_sync_media_setting* setting,
+                                pjmedia_av_sync_media** av_sync_media);
+
+
+/**
+ * Remove a media from synchronizer.
+ *
+ * @param av_sync           The media synchronizer.
+ * @param av_sync_media     The media synchronization handle.
+ *
+ * @return                  PJ_SUCCESS on success.
+ */
+PJ_DECL(pj_status_t) pjmedia_av_sync_del_media(
+                                pjmedia_av_sync *av_sync,
+                                pjmedia_av_sync_media *av_sync_media);
+
+
+/**
+ * Update synchronizer about the last presentation timestamp of the specified
+ * media. Normally this function is called when media is producing a frame
+ * to be rendered (e.g: to a video renderer or audio device speaker).
+ * The media may be requested to adjust its playback delay by the number
+ * returned by the function, for example audio stream can adjust delay
+ * using #pjmedia_jbuf_set_min_delay().
+ *
+ * @param av_sync_media     The media synchronization handle.
+ * @param pts               The presentation timestamp.
+ *
+ * @return                  0  : no action is needed, because the media is
```

Comment:
consider returning pj_status_t instead (and put the `adjustment` as `pj_int32_t *` output parameter).
Because if the return value is zero, it may not be obvious whether something is wrong or truly no adjustment is needed.